sidecar/reloader/main.go

```diff
@@ -106,47 +112,71 @@ func main() {
 					if err != nil {
 						log.Fatal(err)
 					}
+					loadUrl := url.URL{
+						Scheme: "http",
+						Host:   fmt.Sprintf("%s:%d", host, port),
+					}
 					if sha, ok := fileCheck[event.Name]; ok {
 						if !bytes.Equal(sha, getMD5(event.Name)) {
-							log.Println("file changed:", event.Name)
+							log.Printf("%s has changed:\n", event.Name)
 							fileCheck[event.Name] = getMD5(event.Name)
-
-							_, fileName := filepath.Split(event.Name)
-							name := strings.TrimSuffix(fileName, ".conf")
-
-							url := url.URL{
-								Scheme: "http",
-								Host:   fmt.Sprintf("%s:%d", host, port),
-								Path:   fmt.Sprintf("/api/v4/plugins/%s/reload", name),
+							dir, fileName := filepath.Split(event.Name)
+							if dir == fmt.Sprintf("%s/", pluginConfigDir) {
+								pluginName := strings.TrimSuffix(fileName, ".conf")
+								/*
+									curl -i --basic -u admin:public -X PUT
+									"http://localhost:8081/api/v4/nodes/emqx@127.0.0.1/plugins/emqx_delayed_publish/reload"
+								*/
+								loadUrl.Path = fmt.Sprintf("/api/v4/plugins/%s/reload", pluginName)
+								requestWebhook(loadUrl, "PUT", nil, username, password)
+							} else if dir == licenseFileDir {
```

Comment:
How about like this: 
```
if event.Name == licenseFilePath {
// update license
} else {
// reload plugins
}
```

and `licenseFileDir` can be delete

Reply:
no matter, th next version of the api is directly bound to the file